### DISCUSSION TITLE
Converting from yahoo query to yfinance formats

### DISCUSSION THREAD

**Discussion by asafravid** [ranaroussi/yfinance#1454] [Development]
Preliminary (but not only) for `balanceSheetHistory`:

                balanceSheetHistoryYearly = symbol.all_modules[stock_data.symbol]['balanceSheetHistory']
                balanceSheetHistoryQuarterly = symbol.all_modules[stock_data.symbol]['balanceSheetHistoryQuarterly']
                defaultKeyStatistics = symbol.all_modules[stock_data.symbol]['defaultKeyStatistics']
                assetProfile = symbol.all_modules[stock_data.symbol]['assetProfile']
                incomeStatementHistoryYearly = symbol.all_modules[stock_data.symbol]['incomeStatementHistory']
                incomeStatementHistoryQuarterly = symbol.all_modules[stock_data.symbol]['incomeStatementHistory']

                earnings_yearly_yq = []
                earnings_quarterly_yq = []

                total_current_assets_yearly_yq = []
                total_assets_yearly_yq = []
                total_current_liabilities_yearly_yq = []
                total_liabilities_yearly_yq = []
                other_current_assets_yearly_yq = []
                other_current_liabilities_yearly_yq = []
                other_assets_yearly_yq = []

                total_current_assets_quarterly_yq = []
                total_assets_quarterly_yq = []
                total_current_liabilities_quarterly_yq = []
                total_liabilities_quarterly_yq = []
                other_current_assets_quarterly_yq = []
                other_current_liabilities_quarterly_yq = []
                other_assets_quarterly_yq = []

                balance_sheets_yearly_yq = {}
                balance_sheets_quarterly_yq = {}

                for list_element_dict in balanceSheetHistoryYearly['balanceSheetStatements']:
                    balance_sheets_yearly_yq[list_element_dict['endDate']] = {}
                    if 'retainedEarnings'        in list_element_dict:
                        earnings_yearly_yq.append(list_element_dict['retainedEarnings'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['retainedEarnings'] = list_element_dict['retainedEarnings']
                    if 'totalCurrentAssets'      in list_element_dict:
                        total_current_assets_yearly_yq.append(list_element_dict['totalCurrentAssets'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['totalCurrentAssets'] = list_element_dict['totalCurrentAssets']
                    if 'totalAssets'             in list_element_dict:
                        total_assets_yearly_yq.append(list_element_dict['totalAssets'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['totalAssets'] = list_element_dict['totalAssets']
                    if 'totalCurrentLiabilities' in list_element_dict:
                        total_current_liabilities_yearly_yq.append(list_element_dict['totalCurrentLiabilities'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['totalCurrentLiabilities'] = list_element_dict['totalCurrentLiabilities']
                    if 'totalLiab'               in list_element_dict:
                        total_liabilities_yearly_yq.append(list_element_dict['totalLiab'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['totalLiab'] = list_element_dict['totalLiab']
                    if 'otherCurrentAssets'      in list_element_dict:
                        other_current_assets_yearly_yq.append(list_element_dict['otherCurrentAssets'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['otherCurrentAssets'] = list_element_dict['otherCurrentAssets']
                    if 'otherCurrentLiab'        in list_element_dict:
                        other_current_liabilities_yearly_yq.append(list_element_dict['otherCurrentLiab'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['otherCurrentLiab'] = list_element_dict['otherCurrentLiab']
                    if 'otherAssets'             in list_element_dict:
                        other_assets_yearly_yq.append(list_element_dict['otherAssets'])
                        balance_sheets_yearly_yq[list_element_dict['endDate']]['otherAssets'] = list_element_dict['otherAssets']

                for list_element_dict in balanceSheetHistoryQuarterly['balanceSheetStatements']:
                    balance_sheets_quarterly_yq[list_element_dict['endDate']] = {}
                    if 'retainedEarnings'        in list_element_dict:
                        earnings_quarterly_yq.append(list_element_dict['retainedEarnings'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['retainedEarnings'] = list_element_dict['retainedEarnings']
                    if 'totalCurrentAssets'      in list_element_dict:
                        total_current_assets_quarterly_yq.append(list_element_dict['totalCurrentAssets'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['totalCurrentAssets'] = list_element_dict['totalCurrentAssets']
                    if 'totalAssets'             in list_element_dict:
                        total_assets_quarterly_yq.append(list_element_dict['totalAssets'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['totalAssets'] = list_element_dict['totalAssets']
                    if 'totalCurrentLiabilities' in list_element_dict:
                        total_current_liabilities_quarterly_yq.append(list_element_dict['totalCurrentLiabilities'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['totalCurrentLiabilities'] = list_element_dict['totalCurrentLiabilities']
                    if 'totalLiab'               in list_element_dict:
                        total_liabilities_quarterly_yq.append(list_element_dict['totalLiab'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['totalLiab'] = list_element_dict['totalLiab']
                    if 'otherCurrentAssets'      in list_element_dict:
                        other_current_assets_quarterly_yq.append(list_element_dict['otherCurrentAssets'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['otherCurrentAssets'] = list_element_dict['otherCurrentAssets']
                    if 'otherCurrentLiab'        in list_element_dict:
                        other_current_liabilities_quarterly_yq.append(list_element_dict['otherCurrentLiab'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['otherCurrentLiab'] = list_element_dict['otherCurrentLiab']
                    if 'otherAssets'             in list_element_dict:
                        other_assets_quarterly_yq.append(list_element_dict['otherAssets'])
                        balance_sheets_quarterly_yq[list_element_dict['endDate']]['otherAssets'] = list_element_dict['otherAssets']

**Comment by asafravid** [comment]
This is used and tested here:

https://github.com/asafravid/sss/commit/db94b86ad2a34c8b9e7e73b840f7697bc2b24cf6

**Comment by ValueRaider** [comment]
@asafravid what exactly is the purpose of this code? How is it helpful?

**Comment by asafravid** [comment]
Consider a project which uses the particular `dict` formats of yfinance, including functions receiving those parameters, I suggest here a working method to keep the yfinance data structures and providing a conversion/wrapper from yahoo query. It is useful when one is reluctant (time-wise etc) to convert the whole project to use only yahoo query. My 2 cents.
